src/Operation/ModifyCollection.php

```diff
@@ -0,0 +1,146 @@
+     *    used for the returned command result document.
+     *
+     *  * writeConcern (MongoDB\Driver\WriteConcern): Write concern.
+     *
+     *    This is not supported for server versions < 3.2 and will result in an
+     *    exception at execution time if used.
+     *
+     * @param string       $databaseName       Database name
+     * @param string       $collectionOrView   Collection or view to modify
+     * @param string       $collModOptionValue Options and values for collMod
+     * @param array        $options            Command options
+     * @throws InvalidArgumentException for parameter/option parsing errors
+     */
+    public function __construct($databaseName, $collectionOrView, $collModOptionValue, array $options = [])
+    {
+        if (isset($options['session']) && ! $options['session'] instanceof Session) {
+            throw InvalidArgumentException::invalidType('"session" option', $options['session'], 'MongoDB\Driver\Session');
+        }
+
+        if (isset($options['typeMap']) && ! is_array($options['typeMap'])) {
+            throw InvalidArgumentException::invalidType('"typeMap" option', $options['typeMap'], 'array');
+        }
+
+        if (isset($options['writeConcern']) && ! $options['writeConcern'] instanceof WriteConcern) {
+            throw InvalidArgumentException::invalidType('"writeConcern" option', $options['writeConcern'], 'MongoDB\Driver\WriteConcern');
+        }
+
+        if (isset($options['writeConcern']) && $options['writeConcern']->isDefault()) {
+            unset($options['writeConcern']);
+        }
+
+        $this->databaseName = (string) $databaseName;
+        $this->collectionOrView = (string) $collectionOrView;
+        $this->collModOptionValue = $collModOptionValue;
+        $this->options = $options;
+    }
+
+    /**
+     * Execute the operation.
+     *
+     * @see Executable::execute()
+     * @param Server $server
+     * @return array|object Command result document
+     * @throws DriverRuntimeException for other driver errors (e.g. connection errors)
+     */
+    public function execute(Server $server)
+    {
+        if (isset($this->options['writeConcern']) && ! \MongoDB\server_supports_feature($server, self::$wireVersionForWriteConcern)) {
+            throw UnsupportedException::writeConcernNotSupported();
+        }
+
+        $cursor = $server->executeWriteCommand($this->databaseName, $this->createCommand(), $this->createOptions());
+
+        if (isset($this->options['typeMap'])) {
+            $cursor->setTypeMap($this->options['typeMap']);
+        }
+
+        return current($cursor->toArray());
+    }
+
+    /**
+     * Create the collMod command.
+     *
+     * @return Command
+     */
+    private function createCommand()
+    {
+        $cmd = ['collMod' => $this->collectionOrView];
+        foreach($this->collModOptionValue as $key => $value) {
+            $cmd[$key] = $value;
+        }
```

Comment:
If `$this->collectionOptions` is going to be an array, we can simplify this entire method as:

```php
return new Command(['collMod' => $this->collectionName] + $this->collectionOptions);
```

That effectively merges the arrays with a union, ensuring the `collMod` in the left-hand argument remains as-is.

You may decide to just do this in the `execute()` method for forgo `createCommand()` entirely.